sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -236,63 +243,51 @@ void configureLoggerField(FieldDeclaration field, String serviceInterfaceShortNa
-        boolean isContentTypeSetInHeaders) {
-
+    boolean configureBodyWithContentType(BlockStmt body, TypeMirror parameterType, String contentType,
+        String parameterName, boolean isContentTypeSetInHeaders, Elements elementUtils, Types typeUtils) {
+        boolean isSerializationFormatSet;
         if (parameterType == null) {
             // No body content to set
             body.addStatement(
                 StaticJavaParser.parseStatement("httpRequest.getHeaders().set(HttpHeaderName.CONTENT_LENGTH, \"0\");"));
             return false;
         }
 
-        if (contentType == null || contentType.isEmpty()) {
-            // TODO (alzimmer): Why is String octet-stream?
-            contentType = ("byte[]".equals(parameterType) || "String".equals(parameterType))
-                ? APPLICATION_OCTET_STREAM
-                : APPLICATION_JSON;
-        }
+        contentType = determineContentType(contentType, parameterType, elementUtils, typeUtils);
 
         // Set the content type header if it is not already set in the headers
         if (!isContentTypeSetInHeaders) {
-            body.addStatement(StaticJavaParser
-                .parseStatement("httpRequest.getHeaders().set(HttpHeaderName.CONTENT_TYPE, \"" + contentType + "\");"));
-        }
-
-        if ("io.clientcore.core.models.binarydata.BinaryData".equals(parameterType)) {
-            body.tryAddImportToParentCompilationUnit(BinaryData.class);
-            body.addStatement(StaticJavaParser.parseStatement("BinaryData binaryData = " + parameterName + ";"));
-            body.addStatement(StaticJavaParser.parseStatement("if (binaryData.getLength() != null) {"
-                + "httpRequest.getHeaders().set(HttpHeaderName.CONTENT_LENGTH, String.valueOf(binaryData.getLength()));"
-                + "httpRequest.setBody(binaryData); }"));
-            return false;
+            body.addStatement(StaticJavaParser.parseStatement(
+                String.format("httpRequest.getHeaders().set(HttpHeaderName.CONTENT_TYPE, \"%s\");", contentType)));
         }
 
-        if ("byte[]".equals(parameterType)) {
-            body.addStatement(StaticJavaParser
-                .parseStatement("httpRequest.setBody(BinaryData.fromBytes((byte[]) " + parameterName + "));"));
-        } else if ("String".equals(parameterType)) {
-            body.addStatement(StaticJavaParser
-                .parseStatement("httpRequest.setBody(BinaryData.fromString((String) " + parameterName + "));"));
-        } else if ("ByteBuffer".equals(parameterType)) {
-            body.tryAddImportToParentCompilationUnit(ByteBuffer.class);
-            body.addStatement(StaticJavaParser.parseStatement(
-                "httpRequest.setBody(BinaryData.fromBytes(((ByteBuffer) " + parameterName + ").array()));"));
+        if (parameterType.getKind().isPrimitive()) {
+            boolean isJson = Arrays.stream(contentType.split(";"))
```

Comment:
Let's evaluate this further as I know this was copied directly from `RestProxyBase` and it has a comment on whether we should even be doing this. Same with the whole unknown `Content-Type` for `String` defaults to `application/octet-stream`.

Or, more holistically, we should evaluate everything that feels questionable in `RestProxy` / `SwaggerMethodParser` before adding it here to annotation processing. As some of these old choices may have been wrong or no longer applicable and we won't want to copy that.